core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java

```diff
@@ -422,7 +423,12 @@ public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exc
             return HttpResponse.of(HttpStatus.METHOD_NOT_ALLOWED);
         }
 
-        return HttpResponse.from(updateHandler.handle(ctx, req).thenApply(updateResult -> {
+        return HttpResponse.from(updateHandler.handle(ctx, req).handle((updateResult, cause) -> {
+            if (cause != null) {
+                cause = Exceptions.peel(cause);
+                return HttpResponse.ofFailure(cause);
```

Comment:
Q) I understand that even with this change users won't be able to mutate response headers fluently from decorators - I want to make sure this was intended (no problem since I don't think the behavior is changed)

Reply:
It looks cruft. Let me revert this change. 😅